### PR TITLE
Reorganization and Migration of README and internal docs

### PR DESCRIPTION
**Pull request by ravindk89**
## Description

As part of internal discussions, we want to do two big things:

1) Simplify the README and guide users to our Documentation, which is far more in-depth and has better overall coverage
2) Move the `/docs` folder to an internal repository to be a resource for engineers, as the existing content is already generally better suited for ongoing development of MinIO

## Motivation and Context

The `/docs` folder has been engineer-maintained since MinIO's inception. However, it's not a priority for maintenance, cleanups, or updates - it's entirely driven by whether or not an engineer has the time and resources to make meaningful updates.

Furthermore, some content is designed for an audience that is already very familiar with MinIO and S3. This leads to some friction where users turn to the Github docs first, and the web docs second.

By moving the `/docs` folder to an internal repository, we can continue using it as a knowledge base for guiding ongoing development and engineering while keeping a clean and simple README for getting users to the most basic form of usage.

## How to test this PR?

The big test to come will be whether or not to submodule out the internal docs directory.
If we go in that direction, I'll update this part of the PR with instructions on how to init and update a submodule, and push updates and commits as part of a submodule.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [X] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,266 +1,166 @@
...
-Standalone MinIO servers are best suited for early development and evaluation. Certain features such as versioning, object locking, and bucket replication
-require distributed deploying MinIO with Erasure Coding. For extended development and production, deploy MinIO with Erasure Coding enabled - specifically,
-with a *minimum* of 4 drives per MinIO server. See [MinIO Erasure Code Overview](https://min.io/docs/minio/linux/operations/concepts/erasure-coding.html)
-for more complete documentation.
-
-### Stable
-
-Run the following command to run the latest stable image of MinIO as a container using an ephemeral data volume:
+MinIO generally supports deployment on 64-bit Linux or Kubernetes-based infrastructure.
+The fastest way to start experimenting with MinIO is to run it as a container on local machine.
```
Comment (djwfyi) [inline README.md:12]: > ... on **a** local machine.

File: README.md [hunk 2]
```diff
@@ -1,266 +1,166 @@
...
+You can pass the ``-dt`` parameters to run the MinIO container in a detached state with TTY.
 
-You can also connect using any S3-compatible tool, such as the MinIO Client `mc` commandline tool. See
-[Test using MinIO Client `mc`](#test-using-minio-client-mc) for more information on using the `mc` commandline tool. For application developers,
-see <https://min.io/docs/minio/linux/developers/minio-drivers.html> to view MinIO SDKs for supported languages.
+Once started, you can perform object storage operations by pointing any [S3-compatible SDK](https://min.io/docs/minio/linux/developers/minio-drivers.html?ref=github-minio) to ``localhost:9000``.
+Alternatively, open ``localhost:9090`` in your preferred browser to use the [MinIO Console](https://min.io/docs/minio/linux/administration/minio-console.html?ref=github-minio).
 
-> NOTE: To deploy MinIO on with persistent storage, you must map local persistent directories from the host OS to the container using the `podman -v` option. For example, `-v /mnt/data:/data` maps the host OS drive at `/mnt/data` to `/data` on the container.
+For more complete documentation on installing and deploying MinIO, see the following:
```
Comment (djwfyi) [inline README.md:30]: I'd add an h2 just above this:

> `## Official Documentation`

> `### Production Deployments'

   Links to prod quickstarts

> `### Local Testing Deployments`

   Links to OS quickstarts.

File: README.md [hunk 3]
```diff
@@ -1,266 +1,166 @@
...
 
-For hosts with iptables enabled (RHEL, CentOS, etc), you can use `iptables` command to enable all traffic coming to specific ports. Use below command to allow
-access to port 9000
+# Upgrades
 
-```sh
-iptables -A INPUT -p tcp --dport 9000 -j ACCEPT
-service iptables restart
-```
+Community users should follow within 6 months of the latest stable version of MinIO for best results.
```
Comment (djwfyi) [inline README.md:126]: Awkward sentence. Maybe;

> MinIO recommends that Community users alway use the latest version of MinIO for access to the latest features, performance enhancements, and security improvements. As a best practice, Community users should use a release within six months of the latest stable version of MinIO for best results.

File: README.md [hunk 4]
```diff
@@ -1,266 +1,166 @@
...
+You can perform MinIO upgrades by doing a binary replacement without taking the cluster down.
+Once you have updated all MinIO server nodes with the new binary, restart the cluster and check for normal operations.
+MinIO **does not** support "rolling upgrade" strategies operating on a single node at a time.
+Upgrade and restart all nodes in parallel as per our guidelines.
+MinIO restarts typically complete within seconds, and S3 SDK retry functionality enables minimal disruption to client applications.
 
-### Test using MinIO Console
+|**IMPORTANT**|
+|---------------|
+| Always test upgrades in a lower environment (DEV, QA, UAT) before applying to production. Always read the MinIO release notes before performing any upgrade. Performing blind upgrades in production environments carries significant risk. <br> <br> MinIO does not require following tightly to the latest release, and you can selectively upgrade when specific features, fixes, or security patches relevant to your deployment become available. |
```
Comment (djwfyi) [inline README.md:140]: I'd make "release notes" a link to https://github.com/minio/minio/releases.

File: README.md [hunk 5]
```diff
@@ -1,266 +1,166 @@
...
-mc admin update <minio alias, e.g., myminio>
-```
+For Kubernetes environments using the [MinIO Operator](https://min.io/docs/minio/kubernetes/upstream/operations/installation.html), you should update managed Tenants through the [Operator](https://min.io/docs/minio/kubernetes/upstream/operations/install-deploy-manage/upgrade-minio-tenant.html)
 
-- For deployments without external internet access (e.g. airgapped environments), download the binary from <https://dl.min.io> and replace the existing MinIO binary let's say for example `/opt/bin/minio`, apply executable permissions `chmod +x /opt/bin/minio` and proceed to perform `mc admin service restart alias/`.
+# Support
 
-- For installations using Systemd MinIO service, upgrade via RPM/DEB packages **parallelly** on all servers or replace the binary lets say `/opt/bin/minio` on all nodes, apply executable permissions `chmod +x /opt/bin/minio` and process to perform `mc admin service restart alias/`.
+MinIO is released under GNU Affero General Public License v3.0. 
+Community users can receive free best-effort assistance through the [MinIO Community Slack]() or the MinIO Github through [Issues]() or [Discussions]().
```
Comment (djwfyi) [inline README.md:147]: Missing the URLs for these three links.

https://slack.min.io
https://github.com/minio/minio/issues
https://github.com/minio/minio/discussions

File: README.md (hunk 4 above)
Comment (djwfyi) [reply README.md:140]: The second sentence here may conflict with my recommendation above on ln 126.

File: README.md (hunk 3 above)
Comment (djwfyi) [reply README.md:126]: These may conflict with the second sentence on ln 140.

File: README.md (hunk 2 above)
Comment (djwfyi) [reply README.md:30]: Might also consider moving all of these links to the bottom with the Explore Further links.
Though I think we need _some_ link to docs site(s) close to the top of this.

**Review by djwfyi — CHANGES_REQUESTED** [review]
I think this is a good restructuring of this page.
I think the docs links can use some headings to help with scannability.

We should work on the recommendation for upgrading. Is it within 6 months? Or is it "at your convenience to take advantage of features and other updates"? We offer both at the moment in this doc.

**Comment by harshavardhana** [comment]
@ravindk89 I do not think we are ready for this yet, there are many things that we need from `docs/` folder that are not present in the newer docs.

We should also keep some of these developer-centric docs as well, with code and examples.

**Comment by ravindk89** [comment]
@harshavardhana @klauspost what about a middle-point:

- We revise the README.md as suggested
- We cull those sections which have been migrated, linking to the appropriate docs

For developer-centric docs, we could keep those in the private repository and then submodule it here - or keep it entirely public. @abperiasamy for additional insight.

**Comment by klauspost** [comment]
@ravindk89 

> For developer-centric docs, we could keep those in the private repository and then submodule it here - or keep it entirely public.

1. We are OSS. Keeping developer information private is pointless at best. 
2. Implementation specific extensions should be fully public - otherwise why should we have them.
3. Git Submodules are a nightmare.

**Comment by harshavardhana** [comment]
I will close this for now, as we are not there yet and we need some of the markdowns in the docs/ folder

**Pull request closed without merging**